### ISSUE TITLE
NumericUpDown not showing numeric-only keyboard on android

### ISSUE THREAD

**Issue by MaxwellDAssistek** [AvaloniaUI/Avalonia#15850]
### Describe the bug

When using a NumericUpDown control, Avalonia incorrectly shows the full virtual keyboard instead of showing a numeric only virtual keyboard.
<details>
<summary>Screenshot</summary>

![image](https://github.com/AvaloniaUI/Avalonia/assets/61429558/70f797c4-f680-4faa-905f-4dff542308d2)

</details>

### To Reproduce

`<NumericUpDown />`

### Expected behavior

Tapping the NumericUpDown control should present the numeric-only virtual keyboard.

### Avalonia version

11.0.10

### OS

Android

**Comment by stevemonaco** [comment]
Do either of the following work?
```xml
<NumericUpDown TextInputOptions.ContentType="Numeric" />
```
Or
```xml
<Style Selector="NumericUpDown /template/ TextBox#PART_TextBox">
    <Setter Property="(TextInputOptions.ContentType)" Value="Numeric" />
</Style>
```

This seems like it should be default behavior, so leave the issue open for maintainers to evaluate unless it's fixed in nightly.

**Comment by maxkatz6** [comment]
Yes. Current behavior is expected. You need to set TextInputOptions.ContentType if you want to make an effect on a keyboard type.

But it also makes sense to override default value on the NumericUpDown.

**Comment by MaxwellDAssistek** [comment]
`TextInputOptions.ContentType="Numeric"` does work, but it would be nice if `NumericUpDown` had a sensible default.

**Comment by timunie** [comment]
@MaxwellDAssistek as Max said we are open for a change here of the default. In case you want to file a PR for it.

**Comment by MaxwellDAssistek** [comment]
I'd like to also note that `TextInputOptions` does not show up in the attribute autocomplete (at least in Rider). So it is very challenging to figure this out.

**Comment by stevemonaco** [comment]
`TextInputOptions` sets attached properties, so the tooling autocomplete is more difficult than it is with dependency properties. It's somewhat analogous to extension methods where any class could "add" members (methods or metadata properties) to another.
